scenario_player/runner.py

```diff
@@ -276,34 +275,23 @@ def _setup(self, mint_gas, node_count, node_addresses, fund_tx, node_starter, gr
         )
         log.info("Setup done")
         # Start root task
-        root_task_greenlet = gevent.spawn(self.root_task)
-        root_task_greenlet.name = "root_task"
-        greenlets.add(root_task_greenlet)
-        try:
-            gevent.joinall(greenlets, raise_error=True)
-        except BaseException as exc:
-            log.exception("Greenlet Error", exception=exc)
-            if not root_task_greenlet.dead:
-                # Make sure we kill the tasks if a node dies
-                root_task_greenlet.kill()
-            raise
+        self.root_task()
```

Comment:
Why don't we need to start this in a greenlet any more?

Reply:
This whole function is already run in a greenlet. I didn't see a need to add an additional greenlet. We could probably have removed that greenlet even without the other changes.